cv32e40s/tests/programs/custom/cov_holes_generic/cov_holes_generic.c

```diff
@@ -0,0 +1,1000 @@
+    volatile uint16_t select  : 1;
+    volatile uint16_t hit     : 1;
+    volatile uint16_t maskmax : 6;
+    volatile uint16_t dmode   : 1;
+    volatile uint16_t type    : 4;
+  } __attribute__((packed)) volatile fields;
+  volatile uint32_t raw;
+} __attribute__((packed)) mcontrol_t;
+
+typedef union {
+  struct {
+    volatile uint16_t load     : 1;
+    volatile uint16_t store    : 1;
+    volatile uint16_t execute  : 1;
+    volatile uint16_t u        : 1;
+    volatile uint16_t s        : 1;
+    volatile uint16_t res_5_5  : 1;
+    volatile uint16_t m        : 1;
+    volatile uint16_t match    : 4;
+    volatile uint16_t chain    : 1;
+    volatile uint16_t action   : 4;
+    volatile uint16_t size     : 4;
+    volatile uint16_t timing   : 1;
+    volatile uint16_t select   : 1;
+    volatile uint16_t hit      : 1;
+    volatile uint16_t vu       : 1;
+    volatile uint16_t vs       : 1;
+    volatile uint16_t res_26_25: 2;
+    volatile uint16_t dmode    : 1;
+    volatile uint16_t type     : 4;
+  } __attribute__((packed)) volatile fields;
+  volatile uint32_t raw;
+} __attribute__((packed)) mcontrol6_t;
+
+typedef union {
+  struct {
+    volatile uint8_t  action     : 6;
+    volatile uint8_t  u          : 1;
+    volatile uint8_t  s          : 1;
+    volatile uint8_t  res_8_8    : 1;
+    volatile uint8_t  m          : 1;
+    volatile uint8_t  res_10_10  : 1;
+    volatile uint8_t  vu         : 1;
+    volatile uint8_t  vs         : 1;
+    volatile uint16_t res_25_13  : 13;
+    volatile uint8_t  hit        : 1;
+    volatile uint8_t  dmode      : 1;
+    volatile uint8_t  type       : 4;
+  } __attribute__((packed)) volatile fields;
+  volatile uint32_t raw;
+} __attribute__((packed)) etrigger_t;
+
+typedef union {
+  struct {
+    volatile uint16_t info      : 16;
+    volatile uint16_t res_31_16 : 16;
+  } __attribute__((packed)) volatile fields;
+  volatile uint32_t raw;
+} __attribute__((packed)) tinfo_t;
+
+typedef union {
+  struct {
+    volatile uint8_t uie   : 1;  //     0
+    volatile uint8_t sie   : 1;  //     1
+    volatile uint8_t wpri  : 1;  //     2
+    volatile uint8_t mie   : 1;  //     3
+    volatile uint8_t upie  : 1;  //     4
+    volatile uint8_t spie  : 1;  //     5
+    volatile uint8_t wpri0 : 1;  //     6
+    volatile uint8_t mpie  : 1;  //     7
+    volatile uint8_t spp   : 1;  //     8
+    volatile uint8_t wpri1 : 2;  // 10: 9
+    volatile uint8_t mpp   : 2;  // 12:11
+    volatile uint8_t fs    : 2;  // 14:13
+    volatile uint8_t xs    : 2;  // 16:15
+    volatile uint8_t mprv  : 1;  //    17
+    volatile uint8_t sum   : 1;  //    18
+    volatile uint8_t mxr   : 1;  //    19
+    volatile uint8_t tvm   : 1;  //    20
+    volatile uint8_t tw    : 1;  //    21
+    volatile uint8_t tsr   : 1;  //    22
+    volatile uint8_t wpri3 : 8;  // 30:23
+    volatile uint8_t sd    : 1;  //    31
+  } volatile clint;
+  volatile uint32_t raw;
+} __attribute__((packed)) mstatus_t;
```

Comment:
This will be superseded by https://github.com/openhwgroup/core-v-verif/pull/1733/files ?

Reply:
Eventually yes